src/csg_tree.cpp

```diff
@@ -566,76 +444,188 @@ void CsgOpNode::BatchUnion() const {
     for (auto &set : disjointSets) {
       if (set.size() == 1) {
         impls.push_back(
-            std::dynamic_pointer_cast<CsgLeafNode>(children_[start + set[0]])
+            std::static_pointer_cast<CsgLeafNode>(children[start + set[0]])
                 ->GetImpl());
       } else {
         std::vector<std::shared_ptr<CsgLeafNode>> tmp;
         for (size_t j : set) {
           tmp.push_back(
-              std::dynamic_pointer_cast<CsgLeafNode>(children_[start + j]));
+              std::dynamic_pointer_cast<CsgLeafNode>(children[start + j]));
```

Comment:
why is this cast dynamic while the one above is static?

Reply:
ah, I should change these to static.